src/v/archival/ntp_archiver_service.cc

```diff
@@ -3133,4 +3281,26 @@ bool ntp_archiver::local_storage_pressure() const {
                 <= eviction_offset.value();
 }
 
+bool ntp_archiver::flush_in_progress() const {
+    return _flush_uploads_offset.has_value();
+}
+
+bool ntp_archiver::current_flush_complete() const {
+    return flush_in_progress()
+           && uploaded_past_offset(_flush_uploads_offset.value());
+}
+
+bool ntp_archiver::uploaded_past_offset(model::offset o) const {
+    vlog(
+      _rtclog.trace,
+      "In uploaded_past_offset(), manifest last "
+      "offset: {}, query offset: {}.",
+      manifest().get_last_offset(),
+      o);
+    return std::min(
+             manifest().get_last_offset(),
+             _parent.archival_meta_stm()->get_last_clean_at())
```

Comment:
There are a number of checks in `max_collectible_offset()` that might end up returning `offset::max()` or `offset::min()`, and I'm running into those cases in my fixture tests. I figured that using this statement here instead of calling `max_collectible_offset()` would be best for expected behaviour. What do you think?

Reply:
That's true. FWIW the conditions under which it gets set to max are properties of the cluster/topic not being normal TS topics. But I don't feel strongly, and this code makes sense as a reader